lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileSystemCacheModule.java

```diff
@@ -0,0 +1,119 @@
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.inject.Scopes.SINGLETON;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.trino.filesystem.alluxio.AlluxioFileSystemCacheConfig.CACHE_DIRECTORIES;
+import static io.trino.filesystem.alluxio.AlluxioFileSystemCacheConfig.CACHE_MAX_PERCENTAGES;
+import static io.trino.filesystem.alluxio.AlluxioFileSystemCacheConfig.CACHE_MAX_SIZES;
+import static java.lang.String.join;
+import static org.weakref.jmx.guice.ExportBinder.newExporter;
+
+public class AlluxioFileSystemCacheModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        configBinder(binder).bindConfig(AlluxioFileSystemCacheConfig.class);
+        binder.bind(CacheStats.class).to(AlluxioCacheStats.class).in(SINGLETON);
+        newExporter(binder).export(CacheStats.class).as(generator -> generator.generatedNameOf(AlluxioCacheStats.class));
+
+        binder.bind(TrinoFileSystemCache.class).to(AlluxioFileSystemCache.class).in(SINGLETON);
+
+        Properties metricProps = new Properties();
+        metricProps.put("sink.jmx.class", "alluxio.metrics.sink.JmxSink");
+        metricProps.put("sink.jmx.domain", "org.alluxio");
+        MetricsSystem.startSinksFromConfig(new MetricsConfig(metricProps));
+    }
+
+    @Provides
+    @Singleton
+    public static AlluxioConfiguration getAlluxioConfiguration(AlluxioFileSystemCacheConfig config)
+    {
+        checkArgument(config.getMaxCacheSizes().isEmpty() ^ config.getMaxCacheDiskUsagePercentages().isEmpty(),
+                "Either %s or %s must be specified", CACHE_MAX_SIZES, CACHE_MAX_PERCENTAGES);
+        int size = config.getMaxCacheSizes().isEmpty() ? config.getMaxCacheDiskUsagePercentages().size() : config.getMaxCacheSizes().size();
+        checkArgument(config.getCacheDirectories().size() == size,
+                "%s and %s must have the same size", CACHE_DIRECTORIES, config.getMaxCacheSizes().isEmpty() ? CACHE_MAX_PERCENTAGES : CACHE_MAX_SIZES);
+        List<DataSize> maxCacheSizes = config.getMaxCacheSizes().isEmpty() ?
+                calculateMaxCacheSizes(config.getMaxCacheDiskUsagePercentages(), config.getCacheDirectories().stream()
+                        .map(directory -> new File(directory).getTotalSpace()).collect(toImmutableList()))
+                : config.getMaxCacheSizes();
+
+        AlluxioProperties alluxioProperties = new AlluxioProperties();
+        alluxioProperties.set(USER_CLIENT_CACHE_ENABLED, true);
+        alluxioProperties.set(USER_CLIENT_CACHE_DIRS, join(",", config.getCacheDirectories()));
+        alluxioProperties.set(USER_CLIENT_CACHE_SIZE, join(",", maxCacheSizes.stream().map(DataSize::toBytesValueString).toList()));
+        alluxioProperties.set(USER_CLIENT_CACHE_PAGE_SIZE, config.getCachePageSize().toBytesValueString());
+        Duration ttl = config.getCacheTTL();
+        if (ttl != null) {
```

Comment:
Please make that field Optional, we typically prefer making things optional rather than nullable